main/src/mill/main/MainModule.scala

```diff
@@ -173,4 +179,34 @@ trait MainModule extends mill.Module{
       }
     }
   }
+
+  /**
+    * Deletes the given targets from the out directory. Providing no targets will clean everything.
+    */
+  def clean(targets: String*) = mill.T.command {
+    val result = Try {
+      if (ammonite.ops.ls(ammonite.ops.pwd).contains(ammonite.ops.pwd / BuildFile)) {
+        targets match {
+          case Nil =>
+            remove(List(ammonite.ops.pwd / OutDir))
+          case _ =>
+            remove(targets
+              .map(_.replace(".", "/"))
```

Comment:
You should use `resolveDestPaths` (https://github.com/lihaoyi/mill/blob/master/core/src/mill/eval/Evaluator.scala#L342) to figure out what targets to clean, to ensure it stays in sync with any future changes